File: apps/desktop/src-tauri/src/provider_inference.rs
pub fn infer_model_provider_id(model: &str) -> Option<&'static str> {
    let normalized = model.trim().to_ascii_lowercase();
    if normalized.is_empty() {
        return None;
    }

    if normalized.starts_with("claude-") || normalized.starts_with("anthropic/claude-") {
        return Some("anthropic");
    }

    if normalized.starts_with("gpt-") || normalized.starts_with("codex-") {
        return Some("openai");
    }

    None
}

File: apps/desktop/src/lib/providerInference.ts
export type ModelProviderId = 'anthropic' | 'openai';

export const inferModelProviderId = (
  model: string | null | undefined
): ModelProviderId | null => {
  const normalized = (model ?? '').trim().toLowerCase();
  if (!normalized) return null;

  if (
    normalized.startsWith('claude-') ||
    normalized.startsWith('anthropic/claude-')
  ) {
    return 'anthropic';
  }

  if (normalized.startsWith('gpt-') || normalized.startsWith('codex-')) {
    return 'openai';
  }

  return null;
};

export const isAnthropicModel = (model: string | null | undefined): boolean =>
  inferModelProviderId(model) === 'anthropic';
